force-app/main/default/classes/ProgramController.cls

```diff
@@ -1,11 +1,40 @@
 public with sharing class ProgramController {
     @TestVisible
     private static ProgramService service = new ProgramService();
+    @TestVisible
+    private static FieldSetService fieldSetService = new FieldSetService();
+    private static final String ENGAGEMENT_FIELD_SET = 'CreateProgramEngagement';
+    private static final String CONTACT_FIELD_SET = 'CreateContact';
+    private static final String PROGRAM_COHORTS_KEY = 'programCohorts';
+    private static final String ENGAGEMENT_FIELD_SET_KEY = 'engagementFieldSet';
+    private static final String CONTACT_FIELD_SET_KEY = 'contactFieldSet';
 
     @AuraEnabled(cacheable=true)
-    public static List<ProgramCohort__c> getProgramCohortsFromProgramId(Id programId) {
+    public static Map<String, Object> getNewEngagementSetup(Id programId) {
+        Map<String, Object> newEngagementSetup = new Map<String, Object>();
```

Comment:
Ahh yes, I was probably being lazy. I will fix that up.

Reply:
@bmuniswamy - ok so I started working on this but now I'm remembering why I chose to leave it in the controller. From my understanding the controller knows what the UI wants and the Service is decoupled from the UI. I think this is ok because it is the controller just mapping the data returned to the UI... @AndersonTarren @ptewson-sfdo thoughts?

Reply:
I'm still diving into this and haven't wrapped my head around all the aspects of this change yet, but I am leaning towards Jenny's perspective - if this arrangement of this map is in service of the needs of the UI, it probably belongs in the controller. 

There's maybe an argument here to move this even further into a view-specific structure with defined behaviors and properties set up for the UI/view itself, but I'm not going to advocate for that without a better understanding of all these updates yet! 

TL;DR I think it probably belongs in the controller, but maybe hinges on the nuance of whether this is intrinsic to how we retrieve the data from the database, or an arrangement that serves the UI.

Reply:
And now I'm reading through more, I think adding more structure to this response might be worth considering - a class dedicated to storing all this information that can be handed back to the UI layer, with properties for cohorts, engagement fields, and contact fields. We might be able to define some other behaviors in there as well - canUpdateContact() or canUpdateProgram() - that might help us keep the design cleaner? I'm still getting my head around the ramifications of all that though. Happy to do a post-lunch (Pac time) call to discuss!

Reply:
I believe we are relying on Salesforce to handle perms in this situation by using the record-edit-form.